examples/src/bin/gio_task.rs

```diff
@@ -0,0 +1,192 @@
+//! # GTask Sample
+//!
+//! This sample demonstrates how to implement a GIO async/finish API based on GTask (see
+//! https://developer.gnome.org/gio/stable/GTask.html) in Rust using the GTask generated bindings.
+//! This can be useful, for example, when porting to Rust some existing C code exposing such an API.
+
+use gtk::glib::subclass;
+use gtk::glib::subclass::prelude::*;
+use gtk::glib::translate::*;
+use gtk::prelude::AsyncResultExt;
+use gtk::prelude::Cast;
+use gtk::prelude::FileExt;
+use gtk::prelude::ToValue;
+use gtk::{gio, glib};
+
+mod imp {
+    use super::*;
+
+    // FileSize is a simple object that will just contain the read file size.
+    // Initially the optional size field will be initialized to None.
+    pub struct FileSize {
+        pub size: std::cell::RefCell<Option<i64>>,
+    }
+
+    impl ObjectSubclass for FileSize {
+        const NAME: &'static str = "FileSize";
+        type ParentType = glib::Object;
+        type Instance = subclass::simple::InstanceStruct<Self>;
+        type Class = subclass::simple::ClassStruct<Self>;
+        type Type = super::FileSize;
+        glib::object_subclass!();
+
+        fn new() -> Self {
+            Self {
+                size: std::cell::RefCell::new(None),
+            }
+        }
+    }
+
+    impl ObjectImpl for FileSize {}
+}
+
+mod ffi {
+    use super::*;
+    pub type FileSize = <imp::FileSize as ObjectSubclass>::Instance;
+
+    /// # Safety
+    ///
+    /// This is the ffi method to asynchronously get the file size. It accepts a callback of type
+    /// GAsyncReadyCallback, that will be invoked when the async operation completes. Typically,
+    /// this callback will invoke the get_file_size_finish method (implemented below) to get the
+    /// Task result and perform some operation with it.
+    #[no_mangle]
+    pub unsafe extern "C" fn get_file_size_async(
```

Comment:
Maybe worth adding some namespacing on the symbol name here lik in C. `my_file_size_get_file_size_async()` or so?